bitcoind-rpc/src/main/scala/org/bitcoins/rpc/serializers/JsonReaders.scala

```diff
@@ -543,4 +544,15 @@ object JsonReaders {
         case "test"    => TestNet3
       }
   }
+
+  implicit object ServiceIdentifierReads extends Reads[ServiceIdentifier] {
+    override def reads(json: JsValue): JsResult[ServiceIdentifier] =
+      json match {
+        case JsString(s) =>
+          JsSuccess(ServiceIdentifier.fromString(s))
```

Comment:
I believe you actually have to call `fromString` inside of `Try(_)` and then match on the result. Currently it will throw instead of returning a `JsError` if it gets an unexpected `String`